zeebe/journal/src/main/java/io/camunda/zeebe/journal/file/Segment.java

```diff
@@ -121,14 +122,14 @@ public long lastIndex() {
    * @throws UncheckedIOException if the operation failed but the segment is live
    */
   @Override
-  public boolean flush() {
+  public boolean flush() throws FlushException {
     final long lastIndex = lastIndex();
 
     try (final var ignored = metrics.observeSegmentFlush()) {
       buffer.force();
     } catch (final UncheckedIOException e) {
       if (isOpen()) {
-        throw e;
+        throw new FlushException(e.getCause());
```

Comment:
:question: Any advantage in using the cause directly instead of the exception itself? I guess we don't lose too much info?